test/Microsoft.AspNetCore.Razor.Evolution.Test/Intermediate/RazorIRAssert.cs

```diff
@@ -203,7 +203,10 @@ public static void CSharpExpression(string expected, RazorIRNode node)
                 var content = new StringBuilder();
                 for (var i = 0; i < cSharp.Children.Count; i++)
                 {
-                    content.Append(((CSharpTokenIRNode)cSharp.Children[i]).Content);
+                    if (cSharp.Children[i] is RazorIRToken token && token.IsCSharp)
```

Comment:
We casted before, lets just cast now and Assert.True(token.IsCSharp) so we know when we make a behavior change in the future.